src/fillTank.test.js

```diff
@@ -2,10 +1,153 @@
+"use strict";
 
-describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+describe("fillTank", () => {
+  const { fillTank } = require("./fillTank");
 
-  it('should ', () => {
+  it(`should be declared`, () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it(`should be full tank if not 'amount'`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    fillTank(customer, 10);
+
+    expect(customer).toEqual({
+      money: 2680,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
+
+  it(`should be full tank if customer want too much 'amount'`, () => {
+    const customer = {
+      money: 2000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 30,
+      },
+    };
+
+    fillTank(customer, 30, 20);
+
+    expect(customer).toEqual({
+      money: 1700,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
+
+  it(`should be fix count 'amount'
+    if 'customer' don't have enough
+    money and without 'amount'`, () => {
+    const customer = {
+      money: 800,
+      vehicle: {
+        maxTankCapacity: 50,
+        fuelRemains: 15,
+      },
+    };
+
+    fillTank(customer, 50);
+
+    expect(customer).toEqual({
+      money: 0,
+      vehicle: {
+        maxTankCapacity: 50,
+        fuelRemains: 31,
+      },
+    });
+  });
 
+  it(`should be fix count 'amount'
+    if 'customer' don't have enough
+    money and with 'amount'`, () => {
+    const customer = {
+      money: 1000,
+      vehicle: {
+        maxTankCapacity: 50,
+        fuelRemains: 0,
+      },
+    };
+
+    fillTank(customer, 50, 30);
+
+    expect(customer).toEqual({
+      money: 0,
+      vehicle: {
+        maxTankCapacity: 50,
+        fuelRemains: 20,
+      },
+    });
   });
 
-  // write tests here
+  it(`should around sum to decimal part`, () => {
+    const customer = {
+      money: 2000,
+      vehicle: {
+        maxTankCapacity: 30,
+        fuelRemains: 0,
+      },
+    };
+
+    fillTank(customer, 23.54, 10);
+
+    expect(customer).toEqual({
+      money: 1764.6,
+      vehicle: {
+        maxTankCapacity: 30,
+        fuelRemains: 10,
+      },
+    });
+  });
+
+  it(`should around litters to hundredth part`, () => {
```

Comment:
The test description 'should around litters to hundredth part' seems to have a typo. Consider revising it to 'should round liters to the nearest hundredth part' for clarity.